ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Iterator.hs

```diff
@@ -273,10 +273,13 @@ newIterator itEnv@IteratorEnv{..} getItEnv registry from to = do
       BlockPoint { atSlot = tipSlot, withHash = tipHash } ->
         case atSlot endPoint `compare` tipSlot of
           -- The end point is < the tip of the ImmutableDB
-          LT -> streamFromImmDB
+          LT -> streamFromImmDB True
           EQ | withHash endPoint == tipHash
                 -- The end point == the tip of the ImmutableDB
-             -> streamFromImmDB
+             -> streamFromImmDB False
+                   -- No need to check twice; moreover, the second check might
```

Comment:
"No need to check twice": indeed, well-spotted.

I'm not sure about the other reason, it sounds exactly like what #917 fixed. I have just noticed that your #773 does not include that PR. Could it be that #917 fixes the problem you are running into?

Reply:
Doh. I had thought 773 does have 917 but I in fact most recently rebased just before 917 was merged. I was worried this would happen: 773 is getting so long lived and onerous to rebase.

I cherry-picked 917's two commits onto 773 and the tests to pass without this PR's commit. So yes: 917 also avoids this `ReadFutureSlotError` exception.

Which means this PR is just a minor optimization.